File: solutions/scala/problem-00001.scala
/*
[minhuang@mocha:scala]$ time scala problem-00001.scala 
233168

real  0m1.442s
user  0m0.252s
sys 0m0.056s
*/
val sum = (1 until 1000).filter(n => n % 3 == 0 || n % 5 == 0).reduceLeft(_ + _)
println(sum)

File: solutions/haskell/problem-00001.hs
{-
[minhuang@mocha:haskell]$ time runghc problem-00001.hs 
233168

real    0m0.305s
user    0m0.268s
sys     0m0.032s
-}
result = sum [x | x <- [1..999], mod x 3 == 0 || mod x 5 == 0]
main = print result
